File: Difficulty: Easy/Geek Jump/geek-jump.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
  
  int fun(int ind, vector<int>& height, vector<int> &dp)
  {
      if(ind == 0) return 0;
      
      if(dp[ind] != -1) return dp[ind];
      
      int first = abs(height[ind] - height[ind - 1]) + fun(ind-1,height,dp);
      int second = INT_MAX;
      if(ind - 2 >= 0)
      {
          second = abs(height[ind]-height[ind-2]) + fun(ind-2,height,dp);
      }
      
      return dp[ind] = min(first,second);
  }
  
    int minimumEnergy(vector<int>& height, int n) {
        // Code here
        vector<int> dp(n,-1);
        return fun(n-1,height,dp);
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        vector<int> arr(N);
        for (int i = 0; i < N; i++) {
            cin >> arr[i];
        }
        Solution obj;
        cout << obj.minimumEnergy(arr, N) << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Geek Jump/geek-jump.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG{
    public static void main(String args[]) throws IOException{
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t-->0){
            int N=sc.nextInt();
            int []arr=new int[N];
            for(int i=0;i<N;i++){
                arr[i]=sc.nextInt();
                
            }
            Solution obj=new Solution();
            int res=obj.minimumEnergy(arr,N);
            System.out.println(res);
            
        
System.out.println("~");
}
    }
}


// } Driver Code Ends


//User function Template for Java

class Solution{
    
    
    public int fun(int ind,int[] arr,int[] dp)
    {
        if(ind == 0)
        {
            return 0;
        }
        
        if(dp[ind] != -1) return dp[ind];
        
        int right = Integer.MAX_VALUE;
        int left = Integer.MAX_VALUE;
        
        if(ind - 1 >= 0)
        {
            left = Math.abs(arr[ind] - arr[ind-1]) + fun(ind - 1, arr,dp);
        }
        
        if(ind - 2 >= 0)
        {
            right = Math.abs(arr[ind] - arr[ind-2]) + fun(ind - 2, arr,dp);
        }
        
        return dp[ind] = Math.min(left,right);
    }
    
    
    public int minimumEnergy(int arr[],int N){
        //code here
        int[] dp = new int[N+1];
        Arrays.fill(dp,-1);
        return fun(N-1,arr,dp);
    }
}